ARGO/src/TransformSystem.cpp

```diff
@@ -0,0 +1,62 @@
+#include "stdafx.h"
+#include "TransformSystem.h"
+
+TransformSystem::~TransformSystem()
+{
+	BaseSystem::~BaseSystem();
+}
+
+void TransformSystem::update(Entity& t_e)
+{
+	TransformComponent* posComp = dynamic_cast<TransformComponent*>(t_e.getComponent(ComponentType::Transform));
+	if (t_e.getComponent(ComponentType::Input) != nullptr)
+	{
+		InputComponent* inputComp = dynamic_cast<InputComponent*>(t_e.getComponent(ComponentType::Input));
+		updateWithInput(posComp, inputComp);
+	}
+	checkBorder(posComp);
+}
+
+void TransformSystem::checkBorder(TransformComponent* t_pos)
+{
+
+	if (t_pos->getPos().x > 825)
+	{
+		t_pos->setX(-25);
+	}
+	else if (t_pos->getPos().x < -25)
+	{
+		t_pos->setX(825);
+	}
+	if (t_pos->getPos().y > 625)
+	{
+		t_pos->setY(-25);
+	}
+	else if (t_pos->getPos().y < -25)
+	{
+		t_pos->setY(625);
+	}
+}
```

Comment:
magic numbers

Reply:
yes, we dont have a way to get window size yet. Once we have utility class/namespace with globals this will get changed. It will probably be changed even more once we have a "moving camera"